components/core/src/clp/string_utils/string_utils.cpp

```diff
@@ -129,6 +129,23 @@ bool is_wildcard(char c) {
     return false;
 }
 
+void
+replace_unescaped_char(char escape_char, char src_char, char target_char, std::string& src_string) {
+    bool escaped = false;
+
+    auto unescaped = [escape_char, src_char, &escaped](char c) {
```

Comment:
I feel that we don't need to expliclity specify what has to be captured.
Compare to a wildcard capture [&], I guess your current version prevents us from modifying the escape_char? 
Wonder if we make argument to be `char const escape_char, char const src_char` and use a [&[ capture.  In that case, can we modify the value (say, do something like escape_char = 'a') in the lambda?